From 5e6b55c6365c492a50bba26df34bdb96da2a82bd Mon Sep 17 00:00:00 2001
From: Matthieu Vanin <matthieu.vanin@nxp.com>
Date: Wed, 29 Mar 2017 18:13:46 -0400
Subject: [PATCH 7/7] dvdlpcmdec: crop buffers with invalid size

This happens when a buffer is incomplete because of a lossy transport
layer like RTP.
---
 gst/dvdlpcmdec/gstdvdlpcmdec.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/gst/dvdlpcmdec/gstdvdlpcmdec.c b/gst/dvdlpcmdec/gstdvdlpcmdec.c
index ba568c1..e1fdf4b 100644
--- a/gst/dvdlpcmdec/gstdvdlpcmdec.c
+++ b/gst/dvdlpcmdec/gstdvdlpcmdec.c
@@ -676,6 +676,15 @@ gst_dvdlpcmdec_handle_frame (GstAudioDecoder * bdec, GstBuffer * buf)
       if (samples < 1)
         goto drop;
 
+      if (size % (channels * 2)) {
+        GST_WARNING_OBJECT (dvdlpcmdec, "Buffer of size %" G_GSIZE_FORMAT
+            " is not a multiple of %d. Droping incomplete sample",
+            size, channels * 2);
+
+        buf = gst_buffer_make_writable (buf);
+        gst_buffer_set_size (buf, samples * channels * 2);
+      }
+
       gst_buffer_ref (buf);
       break;
     }
-- 
1.9.1

